indy-utils/src/main/kotlin/com/luxoft/blockchainlab/hyperledger/indy/helpers/PoolHelper.kt

```diff
@@ -0,0 +1,88 @@
+package com.luxoft.blockchainlab.hyperledger.indy.helpers
+
+import com.luxoft.blockchainlab.hyperledger.indy.utils.EnvironmentUtils
+import org.hyperledger.indy.sdk.pool.InvalidPoolException
+import org.hyperledger.indy.sdk.pool.Pool
+import org.hyperledger.indy.sdk.pool.PoolJSONParameters
+import org.hyperledger.indy.sdk.pool.PoolJSONParameters.OpenPoolLedgerJSONParameter
+import org.hyperledger.indy.sdk.pool.PoolLedgerConfigExistsException
+import java.io.File
+import java.io.IOException
+import java.nio.file.Files
+import java.util.concurrent.ExecutionException
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.*
+import kotlin.math.absoluteValue
+
+
+/**
+ * Helps to manage ledger pool
+ */
+object PoolHelper {
+
+    const val DEFAULT_POOL_NAME = "default_pool"
+
+    /**
+     * Creates (or not if exists) pool ledger files and returns [Pool] object
+     *
+     * If pool already opened, creates a symlink with random name that targets needed pool
+     *
+     * @param genesisFile: [File] - file with genesis transaction
+     * @param poolName: [String] - name of pool
+     * @param poolConfig: [OpenPoolLedgerJSONParameter] - pool config (when you create new pool)
+     * @return: [Pool] - target pool handle
+     */
+    fun getPool(
```

Comment:
We believe that it`s better to provide different open modes = open, create if missing, recreate, create.

Reply:
Will be fixed.

Reply:
Check the IO best practice how to work with such functions, please